### PR TITLE
feat(dashboards) Add ordering to table widgets

### PR DESCRIPTION
**Pull request by markstory**
Add a column in the widgetquery model so we can store the order for table widgets and top5 results when they are added.

Tables can be sorted on any fields in the results. When a field is removed it is also removed as a sort option.

The UI for choosing the order condition isn't finalized, but this gets it roughed in.

Screen Shot 2021-01-29 at 1 36 44 PM

### CODE REVIEW

File: src/sentry/migrations/0155_add_dashboard_query_orderby.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
```
Comment (markstory) [inline src/sentry/migrations/0155_add_dashboard_query_orderby.py:17]: This table is basically empty in production.

**Review by wedamija — COMMENTED** [review]
Migration lgtm

File: src/sentry/api/serializers/rest_framework/dashboard.py [hunk 2]
```diff
@@ -45,33 +46,48 @@ class DashboardWidgetQuerySerializer(CamelSnakeSerializer):
...
     required_for_create = {"fields", "conditions"}
 
     validate_id = validate_id
 
-    def validate_fields(self, fields):
-        snuba_filter = get_filter("")
+    def validate(self, data):
+        if not data.get("id"):
+            keys = set(data.keys())
+            if keys.intersection(self.required_for_create) != self.required_for_create:
```
Comment (Zylphrex) [inline src/sentry/api/serializers/rest_framework/dashboard.py:58]:
```suggestion
            if not self.required_for_create - keys:
```
🏌️

File: src/sentry/api/serializers/rest_framework/dashboard.py [hunk 3]
```diff
@@ -45,33 +46,48 @@ class DashboardWidgetQuerySerializer(CamelSnakeSerializer):
...
-        if not data.get("id"):
-            keys = set(data.keys())
-            if keys.intersection(self.required_for_create) != self.required_for_create:
-                raise serializers.ValidationError("The fields and conditions fields are required")
+            raise serializers.ValidationError({"fields": f"Invalid fields: {err}"})
         return data
 
+    def _get_attr(self, data, attr, empty_value=None):
+        value = data.get(attr)
+        if value:
```
Comment (Zylphrex) [inline src/sentry/api/serializers/rest_framework/dashboard.py:85]: Should this be an explicit `if value is not None`? Take `conditions` for example, I believe the falsy value `""` is valid.

**Review by dashed — APPROVED** [review]

**Pull request merged**